### PR TITLE
Fix visualization for subject with many images

### PR DESCRIPTION
**Pull request by fepegar**
As reported by @cepa995 in #652.

**Description**
Axis sharing was hard-coded to use columns, but rows are desired for subject with many images (>2):

```python
import torchio as tio
dataset = tio.datasets.RSNAMICCAI('root')
subject = dataset[0]  # it's subject 00792
transform = tio.Compose([
                tio.ToCanonical(),
                tio.Resample("T2w"),
                tio.RescaleIntensity((-1, 1)),
                tio.OneHot(),
            ])
transform(subject).plot()
```

Before:

Figure_0

After:

Figure_1

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/master/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [x] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [ ] Tests added or modified to cover the changes
- [ ] Integration tests passed locally by running `pytest`
- [ ] In-line docstrings updated
- [ ] Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [ ] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

**Comment by fepegar** [comment]
@all-contributors please add @cepa995 for bug
